### ISSUE TITLE
"Random" dial strategy is not random

### ISSUE THREAD

**Issue by C-Pro** [kshvakov/clickhouse#111]
Here
https://github.com/kshvakov/clickhouse/blob/aeedb7b9d0584f393905447021238b5c6e9c5154/connect.go#L47

It will be just linear order shifted (not starting with hosts[0]).

**Comment by kshvakov** [comment]
Yes, it's round robin, I don't know why I gave the name of the param is random )

**Comment by C-Pro** [comment]
connOpenInOrder is round-robin too, I think :)

**Comment by C-Pro** [comment]
As far as I can see `connOpenRandom` will open next host in list if first it tried is unavailable too. The only difference between the modes is in what host will be selected first.
`connOpenInOrder` will first try `hosts[0]`
`connOpenRandom` will first try `hosts[2%len(hosts)]`

**Comment by kshvakov** [comment]
"connOpenRandom" trying to open connection to the new host for each new connection/transaction. "connOpenInOrder" connect only for the first live host in the list.
